src/java.base/share/classes/sun/security/util/AbstractAlgorithmConstraints.java

```diff
@@ -85,8 +86,12 @@ static boolean checkAlgorithm(Set<String> algorithms, String algorithm,
             throw new IllegalArgumentException("No algorithm name specified");
         }
 
-        if (algorithms.contains(algorithm)) {
-            return false;
+        // Wild card matching
+        for (String p : algorithms) {
+            if (Pattern.compile(p.replace("*", ".*"), Pattern.CASE_INSENSITIVE)
```

Comment:
What is the performance impact of this change in terms of both allocations and duration of compiling and matching this regex on overall handshake?

My sense is that this is going to be very expensive for TLS connection overhead.

Reply:
Good point, thanks! This is just a proof of concept draft at this point. We'll work on optimization if we decide to go with this design.

Reply:
`DisabledAlgorithmConstraints` already caches the results of `checkAlgorithm call`, so we should have no performance impact after the 1st handshake.